src/axom/sidre/examples/sidre_mfem_datacollection_restart.cpp

```diff
@@ -220,15 +220,28 @@ int main(int argc, char* argv[])
   std::string protocol = "sidre_conduit_json";
 #endif
 
+  int num_files = -1;
   app.add_option("--cycle", cycle_to_load)
     ->description("Optional simulation cycle to load")
     ->capture_default_str();
   app.add_option("--protocol", protocol)
     ->description("Optional sidre protocol to use for checkpoints and restarts")
     ->check(axom::CLI::IsMember(sidre_protocols))
     ->capture_default_str();
+  app.add_option("--num_files", num_files)
+    ->description(
+      "Optional flag to set the number of output files for parallel "
+      "simulations (default one output file per rank)")
+    ->capture_default_str();
   CLI11_PARSE(app, argc, argv);
 
+#ifdef EXAMPLE_USES_MPI
+  if(num_files > 0)
+  {
+    dc.SetNumFiles(num_files);
+  }
+#endif
```

Comment:
Suggestion (minor, not required, but possibly helpful):
Would it make sense to have the `num_files` option only appear when `EXAMPLE_USES_MPI` ?

Also, it might make sense to either add a note that `num_files` must be greater than 0, or to add a check that `num_files` is between 1 and `num_ranks`, e.g. using  `CLI::range()` 
(similar to https://github.com/LLNL/axom/blob/790d3254d59fd99cde3a7dfdce1a23304724aea3/src/thirdparty/tests/cli11_smoke.cpp#L49)

Reply:
I asked mpi for the number of procs, and then set the range from 1 to num_procs.

If this is run without mpi, then num_procs will be 1, and it will only accept values in the range from 1 to 1, so the behavior is as the user expects